### PR TITLE
feat(es/react): Support script in automatic runtime

### PR DESCRIPTION
**Pull request by JSerFeng**
**Description:**

Inject `const { jsx: _jsx} = require("react/jsx-runtime")` when sourceType is `script`

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/7119.

### CODE REVIEW

File: crates/swc_ecma_transforms_react/src/jsx/tests.rs [hunk 1]
```diff
@@ -1483,3 +1489,74 @@ fn integration(input: PathBuf) {
         },
     );
 }
+
+#[testing::fixture("tests/script/**/input.js")]
+fn script(input: PathBuf) {
+    let mut output = input.with_file_name("output.js");
+    if !output.exists() {
+        output = input.with_file_name("output.mjs");
```
Comment (kdy1) [inline crates/swc_ecma_transforms_react/src/jsx/tests.rs:1497]: I think this approach is too error-prone

Comment (JSerFeng) [reply crates/swc_ecma_transforms_react/src/jsx/tests.rs:1497]: Do you mean that the inject approach is too error-prone

Comment (kdy1) [reply crates/swc_ecma_transforms_react/src/jsx/tests.rs:1497]: This line. Determining test mode based on thr existence of output

Comment (JSerFeng) [reply crates/swc_ecma_transforms_react/src/jsx/tests.rs]: I deleted .mjs output test for script test

Comment (JSerFeng) [reply crates/swc_ecma_transforms_react/src/jsx/tests.rs]: After lunch, I'll fix conflict

File: crates/swc_ecma_transforms_react/src/jsx/mod.rs [hunk 2]
```diff
@@ -975,113 +1033,115 @@ where
...
+                    })
+                    .collect(),
+                optional: false,
+                type_ann: None,
+            }),
+            // require('react')
+            init: Some(Box::new(Expr::Call(CallExpr {
+                span: DUMMY_SP,
+                callee: Callee::Expr(Box::new(Expr::Ident(Ident {
+                    span: DUMMY_SP,
```
Comment (kdy1) [inline crates/swc_ecma_transforms_react/src/jsx/mod.rs:1126]: This should have `SyntaxContext` with `unresolve_mark` as the only `Mark`

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: crates/swc_ecma_transforms_react/src/jsx/mod.rs (hunk 2 above)
Comment (JSerFeng) [reply crates/swc_ecma_transforms_react/src/jsx/mod.rs]: Thanks, done

**Review by kdy1 — APPROVED** [review]
Thanks! I'll merge this along with other breaking changes.

---

swc-bump:
 - swc_ecma_transforms_react --breaking

**Pull request merged**